sdk_v2/kubeflow/training/api/training_client.py

```diff
@@ -12,7 +12,553 @@
+        namespace: str = utils.get_default_target_namespace(),
+    ):
+        """TrainingClient constructor. Configure logging in your application
+            as follows to see detailed information from the TrainingClient APIs:
+            .. code-block:: python
+                import logging
+                logging.basicConfig()
+                log = logging.getLogger("kubeflow.training.api.training_client")
+                log.setLevel(logging.DEBUG)
+
+        Args:
+            config_file: Path to the kube-config file. Defaults to ~/.kube/config.
+            context: Set the active context. Defaults to current_context from the kube-config.
+            client_configuration: Client configuration for cluster authentication.
+                You have to provide valid configuration with Bearer token or
+                with username and password. You can find an example here:
+                https://github.com/kubernetes-client/python/blob/67f9c7a97081b4526470cad53576bc3b71fa6fcc/examples/remote_cluster.py#L31
+            namespace: Target Kubernetes namespace. If SDK runs outside of Kubernetes cluster it
+                takes the namespace from the kube-config context. If SDK runs inside
+                the Kubernetes cluster it takes namespace from the
+                `/var/run/secrets/kubernetes.io/serviceaccount/namespace` file. By default it
+                uses the `default` namespace.
+        """
+
+        # If client configuration is not set, use kube-config to access Kubernetes APIs.
+        if client_configuration is None:
+            # Load kube-config or in-cluster config.
+            if config_file or not utils.is_running_in_k8s():
+                config.load_kube_config(config_file=config_file, context=context)
+            else:
+                config.load_incluster_config()
+
+        k8s_client = client.ApiClient(client_configuration)
+        self.custom_api = client.CustomObjectsApi(k8s_client)
+        self.core_api = client.CoreV1Api(k8s_client)
+        self.api_client = ApiClient()
+
+        self.namespace = namespace
+
+    # TODO (andreyvelich): Currently, only Cluster Training Runtime is supported.
+    def list_runtimes(self) -> List[types.Runtime]:
+        """List of the available runtimes.
+
+        Returns:
+            List[Runtime]: List of available training runtimes. It returns an empty list if
+                runtimes don't exist.
+
+        Raises:
+            TimeoutError: Timeout to list Runtimes.
+            RuntimeError: Failed to list Runtimes.
+        """
+
+        result = []
+        try:
+            thread = self.custom_api.list_cluster_custom_object(
+                constants.GROUP,
+                constants.VERSION,
+                constants.CLUSTER_TRAINING_RUNTIME_PLURAL,
+                async_req=True,
+            )
+            # TODO (andreyvelich): We should de-serialize runtime into object.
+            # For that, we need to import the JobSet models.
```

Comment:
Will you plan to import JobSet models in the next PR?

https://github.com/kubernetes-sigs/jobset/tree/main/sdk/python/jobset/models

Reply:
Yes, I think we should do it. We just need to have discussion how we want to import the JobSet models to our SDK.

Reply:
I see. Have you seen any problems to just import JobSet python models here?
I'm just wondering the reason why you did not import those.
I'm curious about any problems in the JobSet Python models. I would like to reveal those problems.

Reply:
Sure, let me see if we can import JobSet modules directly from the GitHub.
Do you know if we are planning to export those modules to PyPI: https://github.com/kubernetes-sigs/jobset/tree/main/sdk/python ?

Reply:
@tenzen-y I imported the JobSet models and the de-serialization looks to be working 🎉 
https://github.com/kubeflow/training-operator/blob/9bea4cfaeed9c17833760561e2605bba7509bba2/sdk_v2/kubeflow/training/api/training_client.py#L103-L108